File: my-folder/0853-most-profit-assigning-work/solution.py
class Solution:
    def maxProfitAssignment(self, difficulty: List[int], profit: List[int], worker: List[int]) -> int:
        cost = sorted(zip(difficulty,profit))
        pref = []
        best = 0
        for d,p in cost:
            if p > best: best = p
            pref.append((d,best))
        worker.sort()
        ans = 0
        i = 0
        for w in worker:
            while i < len(pref) and pref[i][0] <= w:
                i+=1
            if i > 0:
                ans+=pref[i - 1][1]
        return ans





        ans = 0
        for work in worker:
            curr = 0
            for c in cost:
                if c[0] <= work:
                    curr = max(curr, c[1])
            ans+=curr
        return ans

File: my-folder/0853-most-profit-assigning-work/solution.cpp
class Solution {
public:
    int maxProfitAssignment(vector<int>& difficulty, vector<int>& profit, vector<int>& worker) {
        vector<pair<int, int>> cost;
    for (int i = 0; i < (int)difficulty.size(); i++) {
        cost.push_back({difficulty[i], profit[i]});
    }

    int ans = 0;
    for (int w : worker) {
        int curr = 0;
        for (auto &c : cost) {
            if (c.first <= w)
                curr = max(curr, c.second);
        }
        ans += curr;
    }
    return ans;
    }
};
